internal/pkg/deploy/cloudformation/stack/transformers.go

```diff
@@ -490,31 +490,29 @@ func (s *LoadBalancedWebService) convertALBListener() (*template.ALBListener, er
 		return nil, nil
 	}
 	var rules []template.ALBListenerRule
-	// build listener rule config from primary rule config from manifest.
-	rule, err := routingRuleConfigConverter{
-		rule:         rrConfig.Main,
-		manifest:     s.manifest,
-		httpsEnabled: s.httpsEnabled,
-	}.convert()
-	if err != nil {
-		return nil, err
-	}
-	rules = append(rules, *rule)
-	aliasesFor, err := convertHostedZone(rrConfig.Main.Alias, rrConfig.Main.HostedZone)
-	if err != nil {
-		return nil, err
-	}
-
-	// TODO: @pbhingre build listener rule config from additional rules from manifest.
-
-	httpRedirect := true
-	if rrConfig.Main.RedirectToHTTPS != nil {
-		httpRedirect = aws.BoolValue(rrConfig.Main.RedirectToHTTPS)
+	var aliasesFor template.AliasesForHostedZone
+	for _, routingRule := range rrConfig.RoutingRules() {
+		httpRedirect := true
+		if routingRule.RedirectToHTTPS != nil {
+			httpRedirect = aws.BoolValue(routingRule.RedirectToHTTPS)
+		}
+		rule, err := routingRuleConfigConverter{
+			rule:            routingRule,
+			manifest:        s.manifest,
+			httpsEnabled:    s.httpsEnabled,
+			redirectToHTTPS: httpRedirect,
+		}.convert()
```

Comment:
clever, i like this pattern. haven't seen it in go much.

Reply:
@efekarakus's mind ❤️